### PR TITLE
INT-498 | Fix CPC unit tests

### PR DESCRIPTION
**Pull request by dwelman**
Implement mocked reverse proxy to re-enable unit and integration tests in CPC

### CODE REVIEW

File: orchestrator/mocks/reverse_proxy_mock.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"net/http"
+	"net/url"
+)
+
+// MockReverseProxy is a mock implementation of a ReverseProxy for unit tests.
+type MockReverseProxy struct {
+	Proxy coolfhir.HttpProxy
+}
+
+func NewMockReverseProxy(target *url.URL, orcaPublicUrl *url.URL, transport http.RoundTripper, allowCaching bool) (*MockReverseProxy, error) {
```
Comment (reinkrul) [inline orchestrator/mocks/reverse_proxy_mock.go:15]: `coolfhir.HttpProxy` implements `ReverseProxy`, so this type only exists for the defaults below?

**Review by reinkrul — COMMENTED** [review]
see comments

File: orchestrator/mocks/reverse_proxy_mock.go (hunk 1 above)
Comment (dwelman) [reply orchestrator/mocks/reverse_proxy_mock.go:15]: Yeah I tried to get this to work across the unit and integ test before realising I shouldn't use it for the integ test, and I think based on how I set it up for the integ test I might be able to do similar for the unit test. Will check

Comment (dwelman) [reply orchestrator/mocks/reverse_proxy_mock.go]: Yeah this was leftovers from when it was genuinely a mock client and not just a wrapper, I have removed it

**Pull request merged**